bindata/network/openshift-sdn/sdn-ovs.yaml

```diff
@@ -35,16 +35,42 @@ spec:
           #!/bin/bash
           set -euo pipefail
           export SYSTEMD_IGNORE_CHROOT=yes
+          # systemctl may not be used in a separate PID namespace to reach
+          # the systemd running in PID 1. Therefore we need to use the dbus API
+          systemctl_has_started(){
+          # If a service has started ExecMainPID mantains the PID even after
+          # the process stopping. If it's non 0 we're guaranteed that it has
+          # started at least once.
+          [[ $( gdbus call \
+            --system \
+            --dest org.freedesktop.systemd1 \
+            --object-path /org/freedesktop/systemd1/unit/"$(svc_encode_name ${1})"_2eservice \
+            --method org.freedesktop.DBus.Properties.Get "org.freedesktop.systemd1.Service" "ExecMainPID"
+            ) != '(<uint32 0>,)' ]]
+          }
+          systemctl_restart(){
+            gdbus call \
+              --system \
+              --dest org.freedesktop.systemd1 \
+              --object-path /org/freedesktop/systemd1/unit/"$(svc_encode_name ${1})"_2eservice \
+              --method org.freedesktop.systemd1.Unit.Restart "replace"
+          }
+          svc_encode_name(){
+            # systemd encodes some characters, so far we only need to encode
+            # the character "-" but there may be more in the future.
+            echo "${1//-/_2d}"
+          }
+
           # Check to see if ovs is provided by the node:
-          if systemctl status ovsdb-server; then
+          if systemctl_has_started ovsdb-server; then
```

Comment:
I'm not going to block this, and I'm aware that this is bash within a yaml file and so, but let me give you a few comments.
* If we're going to be reusing this, it'd be worth to add the functions in bash to another folder and call them within there.
* I see a lot of comments, which I appreciate, within the code, but none on the PR/Commit. Please address that out as well as it makes things easier when reviewing.
* This might also be worth a release note.

Reply:
> If we're going to be reusing this, it'd be worth to add the functions in bash to another folder and call them within there.
That'd be good, but since it's embedded and these are static files injected into CNO, I don't know how could we do that.
If you have any suggestion about how to do that I'm happy to hear about it.

> I see a lot of comments, which I appreciate, within the code, but none on the PR/Commit. Please address that out as well as it makes things easier when reviewing.

Will address that.

> This might also be worth a release note.

The code I've just modified has never been released, this was introduced in 4.6 so none of our customers are actually running this code, there is a release note saying that ovs now runs in systemd, which I believe is sufficient.